link_linux.go

```diff
@@ -1974,9 +1974,17 @@ func parseGretapData(link Link, data []syscall.NetlinkRouteAttr) {
 		case nl.IFLA_GRE_IKEY:
 			gre.OKey = ntohl(datum.Value[0:4])
 		case nl.IFLA_GRE_LOCAL:
-			gre.Local = net.IP(datum.Value[0:16])
+			if len(datum.Value) == 4 {
```

Comment:
This logic is basically saying `use the whole Value buffer as input`. Can you please try replacing it with
```
gre.Local = net.IP(datum.Value)
```
At the end

Reply:
I think @aboch was suggesting that in both cases we want to use the whole buffer, so we can replace the whole `if` statement with just
```
gre.Local = net.IP(datum.Value)
```
There doesn't seem to be any need to slice it at all.